### ISSUE TITLE
Courseplay suddenly asks startup questions and seems to have lost all the courses

### ISSUE THREAD

**Issue by mattwilkins1** [Courseplay/courseplay#1155]
CP is brilliant and has been working well. It's always made loading a saved game very slow, but that's not a problem. 

I loaded a savegame yesterday and it was like when I first loaded CP, asked me about scan fields and paying workers. When I loaded CP in game all the courses were gone. I made some new ones, saved and have had the same problem again. I have a copy of the courseplay.xml and its c. 5mb and seems to be fine. Game Log is below and it looks like there is a problem with the courseplay.xml, but I'm not sure what and how to fix it. 

I've tried will all mods out, same problem and have a sensible number (20) of mods. 

I hope someone can help; FS is unplayable without CP. 

Many thanks in advance

GIANTS Engine Runtime 6.0.2 64bit (Build Date: Feb  5 2015)
Copyright (c) 2008-2014, GIANTS Software GmbH (giants-software.com), All Rights Reserved.
Copyright (c) 2003-2014, Christian Ammann and Stefan Geiger, All Rights Reserved.
Application: FarmingSimulator2015
Main System
  CPU: AMD E1-2100 APU with Radeon(TM) HD Graphics    
  Memory: 11721 MB
  OS: Windows NT 6.2 64-bit
Physics System
  Version: 5.9.4
  Thread(s): 1
Input System
  Keyboard enabled
  Mouse enabled
  Gamepad/Joystick disabled
  Force Feedback disabled
Sound System
  Driver: OpenAL
  Version: 1.1
  Device:  Generic Software
  Max. sources: 256
Render System
  Driver: OpenGL
  Card Vendor: ATI Technologies Inc.
  Renderer: AMD Radeon HD 8210
  Version: 3.2.12615 Compatibility Profile Context 13.250.31.0
  Shader Version: 4.30
    GL_NV_texture_compression_vtc not supported
    max_texture_layers: 16
    OpenGL initialization successful
Hardware Profile
  Level: Medium (auto)
  View Distance Factor: 1.000000
  Shadow Quality: 1.000000
  Skip Mipmaps: 1
  LOD Distance Factor: 1.000000
  Terrain LOD Distance Factor: 1.000000
  Foliage View Distance Factor: 1.000000
  Volume Mesh Tessellation Factor: 1.000000
  Tyre Tracks Segments Factor: 1.000000
Farming Simulator 15 INT
  Version: 1.2.1.0 1.2.1RC11
  Available Languages: en de fr es it pl cz hu nl ru jp pt cs ct br tr ro kr
  Language: en
Game vehicle types loaded
Mod directory: C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/mods/
Load dlc: pdlc_jcbPack
Load mod: AllroadAudi
Load mod: BrantnerDPW18000
Load mod: Brantner_Seed
Load mod: Chevrolet_C10
Load mod: FS15_kuhnVB2190
Load mod: FS15_marshallPack
Load mod: FS15_placeableSheds_MW_mailman
Load mod: FS15_PoettingerJumboCombiline6610
Load mod: HummerH2
Load mod: kw900_v3
Load mod: LS15_ClarkForklift
Load mod: OlbertSW300
Load mod: placeable_fertilizerTank_LS15
Error: Failed to open xml file 'C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/mods//placeable_fertilizerTank_LS15/modDesc.xml'.
Error: Missing descVersion attribute in mod placeable_fertilizerTank_LS15
Load mod: placeable_floodlight_LS15
Load mod: placeable_gasTank_LS15
Load mod: placeable_seedPallets_LS15
Load mod: placeable_strawPowerPlant
Load mod: portableseeds
Load mod: Unimog_U400_WB
Load mod: ZZZ_courseplay
  Register vehicle type: BrantnerDPW18000.dpw18000
  Register vehicle type: Chevrolet_C10.ChevroletC101966
  Register vehicle type: FS15_marshallPack.rotorManureSpreader
  Register vehicle type: FS15_marshallPack.manureSpreaderPlaneMover
  Register vehicle type: HummerH2.HummerH2
  Register vehicle type: LS15_ClarkForklift.forklift
  Register vehicle type: LS15_ClarkForklift.ClarkSchaufel
  Register vehicle type: OlbertSW300.OlbertSW300
### Courseplay: installed into 23 vehicles
data/sky/sky_day_night.i3d (28.95 ms)
data/sky/rain.i3d (4.29 ms)
data/sky/hail.i3d (2.39 ms)
data/sky/dust.i3d (2.12 ms)
data/maps/map02.i3d (35856.40 ms)
## Courseplay: setting up signs
C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/mods//ZZZ_courseplay/img/signs/normal.i3d (2.62 ms)
C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/mods//ZZZ_courseplay/img/signs/cross.i3d (3.11 ms)
C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/mods//ZZZ_courseplay/img/signs/wait.i3d (4.31 ms)
C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/mods//ZZZ_courseplay/img/signs/start.i3d (4.08 ms)
C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/mods//ZZZ_courseplay/img/signs/stop.i3d (3.98 ms)
## Courseplay: setting up fields (basic)
Error: C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/savegame3/courseplay.xml(60180). Error reading end tag..
## Courseplay: loading courses and folders from "courseplay.xml"
Error: C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/savegame3/courseplay.xml(60180). Error reading end tag..
C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/mods//HummerH2/HummerH2.i3d (1435.79 ms)
dataS2/character/player/player.i3d (2597.39 ms)
## Courseplay: Hummer H2: aiTrafficCollisionTrigger missing. Traffic collision prevention will not work!
C:/Program Files (x86)/Steam/steamapps/common/Farming Simulator 15/pdlc/jcbPack/jcb/jcbFastrac3000.i3d (3554.75 ms)
C:/Program Files (x86)/Steam/steamapps/common/Farming Simulator 15/pdlc/jcbPack/jcb/jcbFastrac8000.i3d (3420.50 ms)
data/vehicles/tools/horsch/horschPronto9SW.i3d (195.73 ms)
data/vehicles/particleAnimations/horsch/horschPronto9SW.i3d (15.13 ms)
data/vehicles/trailers/strautmann/strautmannTeraVitesse5201.i3d (103.35 ms)
data/vehicles/trailers/ursus/ursusT127.i3d (53.63 ms)
data/vehicles/steerable/kuhn/kuhnSPVConfort12.i3d (116.32 ms)
C:/Program Files (x86)/Steam/steamapps/common/Farming Simulator 15/pdlc/jcbPack/jcb/jcb260.i3d (3257.91 ms)
## Courseplay: 260: aiTrafficCollisionTrigger missing. Traffic collision prevention will not work!
C:/Program Files (x86)/Steam/steamapps/common/Farming Simulator 15/pdlc/jcbPack/jcb/jcbBaleFork.i3d (1432.32 ms)
data/vehicles/trailers/fliegl/flieglSemitrailer.i3d (62.94 ms)
data/vehicles/tools/kuhn/kuhnFC3525F.i3d (62.02 ms)
Error: Can't load resource 'C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/mods//Brantner_Seed/shared/dirt_normal.dds'.
Error: Can't load resource 'C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/mods//Brantner_Seed/shared/dirt_normal.dds'.
C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/mods//Brantner_Seed/Brantner_Seed.i3d (218.03 ms)
data/vehicles/steerable/cars/piQup.i3d (72.48 ms)
data/vehicles/steerable/newHolland/newHollandCR1090.i3d (243.73 ms)
data/vehicles/steerable/man/man22s.i3d (184.82 ms)
C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/mods//LS15_ClarkForklift/StaplerLS15.i3d (695.59 ms)
data/vehicles/steerable/grimme/grimmeTectron415.i3d (250.90 ms)
data/vehicles/tools/grimme/grimmeGL660.i3d (91.02 ms)
data/vehicles/particleAnimations/grimme/grimmeGL660.i3d (3.20 ms)
data/vehicles/trailers/zunhammer/zunhammer18500PE.i3d (69.23 ms)
data/vehicles/trailers/ursus/ursusN270.i3d (85.97 ms)
data/vehicles/steerable/caseIH/caseIHPuma160.i3d (130.15 ms)
data/vehicles/tools/lemken/lemkenHeliodorGigant101200.i3d (95.83 ms)
data/vehicles/particleAnimations/lemken/lemkenHeliodorGigant101200.i3d (11.41 ms)
data/vehicles/steerable/same/sameFortis190.i3d (149.98 ms)
data/vehicles/steerable/ponsse/ponsseScorpion.i3d (240.07 ms)
## Courseplay: 260: aiTrafficCollisionTrigger missing. Traffic collision prevention will not work!
C:/Program Files (x86)/Steam/steamapps/common/Farming Simulator 15/pdlc/jcbPack/jcb/jcbPalletFork.i3d (201.23 ms)
data/vehicles/trailers/fliegl/flieglDPW180.i3d (59.34 ms)
data/vehicles/tools/damcon/damconPL75.i3d (70.97 ms)
data/vehicles/particleAnimations/damcon/damconPL75.i3d (4.69 ms)
data/vehicles/steerable/ursus/ursus15014.i3d (105.77 ms)
data/vehicles/steerable/lindner/lindnerGeotrac94.i3d (134.44 ms)
data/vehicles/trailers/fuelTrailer/fuelTrailer.i3d (49.38 ms)
data/vehicles/trailers/krampe/krampeSB3060.i3d (88.67 ms)
C:/Program Files (x86)/Steam/steamapps/common/Farming Simulator 15/pdlc/jcbPack/jcb/jcbShovel.i3d (10.13 ms)
data/vehicles/tools/horsch/horschMaestro12SW.i3d (84.13 ms)
data/vehicles/particleAnimations/horsch/horschMaestro12SW.i3d (3.14 ms)
data/vehicles/steerable/grimme/grimmeMaxtron620.i3d (216.94 ms)
C:/Program Files (x86)/Steam/steamapps/common/Farming Simulator 15/pdlc/jcbPack/jcb/jcb435s.i3d (3349.82 ms)
data/vehicles/wheelLoaders/lizard/wheelLoaderShovel.i3d (17.08 ms)
data/vehicles/steerable/newHolland/newHollandT475.i3d (163.70 ms)
data/vehicles/cutters/caseIH/caseIH3162Cutter.i3d (109.86 ms)
C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/mods//kw900_v3/kenworth.i3d (268.10 ms)
data/vehicles/trailers/amazone/amazoneZGB8200.i3d (59.15 ms)
data/vehicles/tools/amazone/amazoneZAM1501.i3d (60.19 ms)
data/vehicles/steerable/liebherr/liebherrTL432_7.i3d (208.30 ms)
data/vehicles/telehandlers/lizard/telehandlerBaleGrab.i3d (16.73 ms)
data/vehicles/cutters/newHolland/newHollandHeader980CF12.i3d (96.17 ms)
C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/mods//OlbertSW300/OlbertSW300.i3d (381.49 ms)
data/vehicles/tools/suer/suerSB1000.i3d (13.17 ms)
data/vehicles/steerable/liebherr/liebherrL538.i3d (198.28 ms)
C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/mods//Chevrolet_C10/ChevroletC101966.i3d (1146.97 ms)
Warning: FillPlaneGeometry shape geometry must be called 'fillVolumeShape', not 'object0'
C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/mods//AllroadAudi/Allroad.i3d (509.06 ms)
data/vehicles/trailers/kotte/garantFieldContainer.i3d (72.76 ms)
data/vehicles/trailers/kotte/garantTSA.i3d (91.68 ms)
data/vehicles/trailers/fliegl/flieglDolly.i3d (24.32 ms)
data/vehicles/tools/vogelNoot/vogelNootHeros1000.i3d (86.70 ms)
data/vehicles/particleAnimations/vogelNoot/vogelNootHeros1000.i3d (8.58 ms)
Warning: FillPlaneGeometry shape geometry must be called 'fillVolumeShape', not 'object0'
data/vehicles/tools/amazone/amazoneUF1801.i3d (68.58 ms)
data/vehicles/tools/amazone/amazoneFT1001.i3d (57.64 ms)
C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/mods//Unimog_U400_WB/U400_WB_street.i3d (2306.30 ms)
data/vehicles/balers/newHolland/newHollandBB1290.i3d (80.50 ms)
data/maps/models/objects/squarebales/baleHay240.i3d (12.02 ms)
data/vehicles/trailers/arcusin/arcusinAutostackFS63-72.i3d (81.16 ms)
C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/mods//Unimog_U400_WB/U400_lang_WB.i3d (657.10 ms)
data/vehicles/tools/kuhn/kuhnGF6502.i3d (68.14 ms)
data/vehicles/steerable/huerlimann/huerlimannH488.i3d (153.19 ms)
data/vehicles/tools/suer/suerSB700.i3d (2.83 ms)
data/vehicles/tools/kuhn/kuhnGA8521.i3d (78.86 ms)
data/vehicles/tools/kuhn/kuhnFC10030.i3d (81.39 ms)
C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/mods//Unimog_U400_WB/U400_WB.i3d (193.18 ms)
C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/mods//Unimog_U400_WB/WB_Plattform.i3d (468.51 ms)
C:/Program Files (x86)/Steam/steamapps/common/Farming Simulator 15/pdlc/jcbPack/jcb/jcbTM320s.i3d (3512.47 ms)
data/vehicles/telehandlers/lizard/telehandlerBaleFork.i3d (3.38 ms)
data/vehicles/telehandlers/lizard/telehandlerShovel.i3d (3.60 ms)
data/vehicles/trailers/fliegl/flieglVFW25000.i3d (63.38 ms)
data/vehicles/balers/newHolland/newHollandRollBelt150.i3d (88.72 ms)
data/maps/models/objects/roundbales/roundbaleHay_w112_d130.i3d (28.21 ms)
data/maps/models/objects/roundbales/roundbaleStraw_w112_d130.i3d (7.74 ms)
data/placeables/highPressureWasher/kaercher/kaercherHDS918-4M.i3d (53.12 ms)
data/placeables/windTurbine/windTurbine.i3d (75.90 ms)
data/placeables/beehive/beehive.i3d (37.99 ms)
data/maps/models/objects/pallets/woolPallet.i3d (6.13 ms)
C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/mods//placeable_gasTank_LS15/gasTankPlaceable.i3d (30.82 ms)
C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/mods//placeable_seedPallets_LS15/seedPalletsPlaceable.i3d (80.57 ms)
C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/mods//FS15_placeableSheds_MW_mailman/unterstand2places.i3d (94.66 ms)
data/objects/pallets/treeSaplingPallet.i3d (9.14 ms)
data/objects/pallets/treeSapling.i3d (2.19 ms)
C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/mods//placeable_strawPowerPlant/powerPlantPlaceable.i3d (185.30 ms)
dataS/character/player/playerCCT.i3d (57.96 ms)
data/firstPerson/chainsaws/husqvarna/husqvarna550.i3d (53.99 ms)
Error: C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/savegame3/courseplay.xml(60180). Error reading end tag..
Error: C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/savegame3/courseplay.xml(60180). Error reading end tag..
Warning: Deleting object 'seedTrigger' before all triggers are removed.
Warning: Deleting object 'fertilizerTrigger' before all triggers are removed.
Warning: Deleting object 'seedTrigger' before all triggers are removed.
Warning: Deleting object 'seedTrigger' before all triggers are removed.
Warning: Deleting object 'seedTrigger' before all triggers are removed.
Warning: Deleting object 'seedTrigger' before all triggers are removed.
Warning: Deleting object 'fertilizerTrigger' before all triggers are removed.
Warning: Deleting object 'fertilizerTrigger' before all triggers are removed.
Warning: Deleting object 'fertilizerTrigger' before all triggers are removed.
Warning: Deleting object 'fertilizerTrigger' before all triggers are removed.
Warning: Deleting object 'seedTrigger' before all triggers are removed.
Warning: Deleting object 'seedTrigger' before all triggers are removed.
Warning: Deleting object 'seedTrigger' before all triggers are removed.
Warning: Deleting object 'seedTrigger' before all triggers are removed.
Warning: Deleting object 'fertilizerTrigger' before all triggers are removed.
Warning: Deleting object 'fertilizerTrigger' before all triggers are removed.
Warning: Deleting object 'fertilizerTrigger' before all triggers are removed.
Warning: Deleting object 'fertilizerTrigger' before all triggers are removed.
Warning: Deleting object 'seedTrigger' before all triggers are removed.
Warning: Deleting object 'seedTrigger' before all triggers are removed.
Warning: Deleting object 'seedTrigger' before all triggers are removed.
Warning: Deleting object 'seedTrigger' before all triggers are removed.
Warning: Deleting object 'fertilizerTrigger' before all triggers are removed.
Warning: Deleting object 'fertilizerTrigger' before all triggers are removed.
Warning: Deleting object 'fertilizerTrigger' before all triggers are removed.
Warning: Deleting object 'fertilizerTrigger' before all triggers are removed.
Warning: Deleting object 'seedTrigger' before all triggers are removed.
Warning: Deleting object 'seedTrigger' before all triggers are removed.
Warning: Deleting object 'seedTrigger' before all triggers are removed.
Warning: Deleting object 'seedTrigger' before all triggers are removed.
Warning: Deleting object 'fertilizerTrigger' before all triggers are removed.
Warning: Deleting object 'fertilizerTrigger' before all triggers are removed.
Warning: Deleting object 'fertilizerTrigger' before all triggers are removed.
Warning: Deleting object 'fertilizerTrigger' before all triggers are removed.

**Comment by gui7545** [comment]
Your file is corrupted : 
```
Error: C:/Users/Lenovo/Documents/My Games/FarmingSimulator2015/savegame3/courseplay.xml(60180). Error reading end tag..
```
delete the courseplay.xml (keep the copy) and it should be good

**Comment by mattwilkins1** [comment]
Thanks, tried that but no joy. Very oddly after opening the game and setting a couple of courses (normal save and exit) the old courseplay xml re-appeared still the same size and content. Deleted all the savegame backups so no clue where it's getting the data from, but each time there it is.

**Comment by VardemisDatara** [comment]
Hello, I have the same problem, the 2 file courseplay use are not save in the savegame. 
Courseplay sometime keep the course but they didn't appear in game but the overfile containing the field data is absent at each time. 
it's really annoying I can't play alone on the huge map and can't run the farm easily :(

**Comment by gui7545** [comment]
I know that there is some trouble with very big files .
Try to not save all your courses, game will load faster

**Comment by FarmerFiero** [comment]
Hello all
I just had this problem too.   Courseplay couldn't read the .Xml and was throwing the "error reading end tag" in the log.   After digging in the .Xml and comparing it to a working .Xml from another map, I noticed that a few lines of code were missing...  I added the missing code and deleted some of my largest courses, and it worked.    Im currently on the Michigan Cash crops map and my courses were just too large to load up.    The lines of code i was missing were just below the last course at the very bottom of the .Xml.    Below is what it should look like.  For some reason or another when saving REALLY large courses, the last 3 lines of code didn't get written. 
      
                            <waypoint412 pos="-1070.65 1411.27" angle="-90.70" speed="0" wait="1" />
                            <waypoint413 pos="-1070.65 1411.27" angle="-90.69" speed="0" crossing="1" />
                   </course>
              </courses>
          </XML>

I hope this helps for anybody else having a similar issue.

**Comment by fck54** [comment]
Le 27/11/2015 04:19, FarmerFiero a écrit :
>
> Hello all
> I just had this problem too. Courseplay couldn't read the .Xml and was 
> throwing the "error reading end tag" in the log. After digging in the 
> .Xml and comparing it to a working .Xml from another map, I noticed 
> that a few lines of code were missing... I added the missing code and 
> deleted some of my largest courses, and it worked. Im currently on the 
> Michigan Cash crops map and my courses were just too large to load up. 
> The lines of code i was missing were just below the last course at the 
> very bottom of the .Xml. Below is what it should look like. For some 
> reason or another when saving REALLY large courses, the last 3 lines 
> of code didn't get written.
>
> |<waypoint412 pos="-1070.65 1411.27" angle="-90.70" speed="0" wait="1" 
> /> <waypoint413 pos="-1070.65 1411.27" angle="-90.69" speed="0" 
> crossing="1" /> </course> </courses> </XML> |
>
> I hope this helps for anybody else having a similar issue.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/Courseplay/courseplay/issues/1155#issuecomment-160033770>.
>
had the same issue, was coming from added mod map, you have to watch for 
errors in the log file

**Comment by valdimellen** [comment]
> FarmerFiero commented on Nov 27, 2015
Hello all
I just had this problem too. Courseplay couldn't read the courseplay.Xml and was throwing the "error reading end tag" in the log. After digging in the courseplay.Xml and comparing it to a working courseplay.Xml from another map, I noticed that a few lines of code were missing... I added the missing code and deleted some of my largest courses, and it worked. Im currently on the Michigan Cash crops map and my courses were just too large to load up. The lines of code i was missing were just below the last course at the very bottom of the courseplay.Xml. Below is what it should look like. For some reason or another when saving REALLY large courses, the last 3 lines of code didn't get written.

                        <waypoint412 pos="-1070.65 1411.27" angle="-90.70" speed="0" wait="1" />
                        <waypoint413 pos="-1070.65 1411.27" angle="-90.69" speed="0" crossing="1" />
               </course>
          </courses>
      </XML>


it worked for me.. THX.
